### PR TITLE
Last Used Login: implement the login option

### PR DESCRIPTION
**Pull request by arcangelini**
## Proposed Changes

* Look for `last_used_authentication_method` cookie
* Render the previously used method button on the left.
* Replace the original location of the button with an email option

## Why are these changes being made?

This is the final (hopefully) installment of the riveting Last Used Authentication Method saga. Up until this point we have structured the components to work better together and autonomously. Now it is time to put it all together to allow users the opportunity to use their last method.

image

## Testing Instructions

Test the login links and verify there is no regression.

### CODE REVIEW

**Comment by agrullon95** [comment]
Some issues I found:

1. The "Continue with Email" doesn't function when I click on it.
2. When I log in with Email, the email login button gets added under "Previously used" on the left side but remains on the right. The "Continue with Email" is missing in this scenario.

File: client/blocks/login/login-form.jsx [hunk 1]
```diff
@@ -857,207 +937,225 @@ export class LoginForm extends Component {
...
-									{
-										args: {
-											email: this.props.socialAccountLinkEmail,
-											service: capitalize( this.props.socialAccountLinkService ),
-										},
-									}
+					{ this.state.lastUsedAuthenticationMethod ? (
+						<>
+							<span style={ { marginBottom: '5px', fontSize: '14px', fontWeight: 600 } }>
+								Previously used
```
Comment (agrullon95) [inline client/blocks/login/login-form.jsx]: Let's translate `Previously used` and move styling to .scss file.

File: client/blocks/login/social.jsx [hunk 2]
```diff
@@ -2,134 +2,103 @@ import { Card } from '@automattic/components';
...
+					responseHandler={ this.props.handleLogin }
+					onClick={ this.props.trackLoginAndRememberRedirect }
+					socialServiceResponse={ this.props.socialServiceResponse }
+					key={ 3 }
+					isLogin
+				/>
+			),
+		},
+		{
+			service: 'magic',
```
Comment (agrullon95) [inline client/blocks/login/social.jsx]: Let's rename this `magic-login`. That should fix the "Continue with Email" button issue where it doesn't appear if you previously logged in with the magic login link option.

```suggestion
			service: 'magic-login'
```

File: client/blocks/login/login-form.jsx [hunk 3]
```diff
@@ -857,207 +937,225 @@ export class LoginForm extends Component {
...
 						<SocialLoginForm
-							linkingSocialService={
-								this.props.socialAccountIsLinking ? this.props.socialAccountLinkService : null
-							}
-							onSuccess={ this.props.onSuccess }
-							socialService={ this.props.socialService }
+							lastUsedAuthenticationMethod={ this.state.lastUsedAuthenticationMethod }
+							handleLogin={ this.handleSocialLogin }
+							trackLoginAndRememberRedirect={ this.trackLoginAndRememberRedirect }
+							reset={ this.resetLastUsedAuthenticationMethod }
```
Comment (agrullon95) [inline client/blocks/login/login-form.jsx]: Updating this prop should fix the "Continue with Email" button:

```suggestion
							resetLastUsedAuthenticationMethod={ this.resetLastUsedAuthenticationMethod }
```

**Comment by arcangelini** [comment]
Thanks @agrullon95 

I pushed kind of late last night and had some oversights. 🤣

**Review by heavyweight — CHANGES_REQUESTED** [review]
I was logged in via github. Logged out and logged in with user/pass.
After logging out I got here..
Screenshot 2024-08-29 at 11 55 55

**Review by agrullon95 — APPROVED** [review]
LGTM! Good job.

I wasn't able to test Apple and Github, but the rest of the login methods are working well.

**Comment by escapemanuele** [comment]
I believe @heavyweight is going to be able to test Apple for sure

**Comment by heavyweight** [comment]
Tested signup and login with apple and didn't find any issues

**Review by escapemanuele — APPROVED** [review]
It is now good to go, after all the things we need to do for the experiment.

Well done!

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/16654813

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Hi @arcangelini, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include this string: `Previously used`

Thank you in advance!

**Pull request merged**